### PR TITLE
Add documentation for Google ML & BigQuery nodes

### PR DESCRIPTION
**Pull request by wesm87**
Also changes "Google Pub/Sub" to "GCP Pub/Sub" to match the new naming
convention

### CODE REVIEW

File: docs/workflows/data/google-big-query.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# GCP BigQuery
```
Comment (cephirothdy2j) [inline docs/workflows/data/google-big-query.md]: Add "Node" to the end of this title to be consistent with other nodes in docs.

File: docs/workflows/data/google-big-query.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+The GCP (Google Cloud Platform) BigQuery node allows a workflow to interact with Google BigQuery by fetching, creating, updating, or deleting data and optionally store the result on the workflow payload.
```
Comment (cephirothdy2j) [inline docs/workflows/data/google-big-query.md]: store => storing (to match tense)

File: docs/workflows/data/google-big-query.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+![GCP BigQuery Node](/images/workflows/data/google-big-query-node.png "GCP BigQuery Node")
+
+## Configuration
+
+Configuration for the node is broken up into four sections.
+
+### Service Account Auth Token
+
+A [service account auth token](https://cloud.google.com/docs/authentication/getting-started#creating_a_service_account) is required for the workflow to authenticate with Google ML. You may enter this token one of two ways:
```
Comment (cephirothdy2j) [inline docs/workflows/data/google-big-query.md]: "Google ML"; guessing that was a copy/paste error.

File: docs/workflows/data/google-big-query.md (hunk 2 above)
Comment (cephirothdy2j) [reply docs/workflows/data/google-big-query.md]: I'd also say "the result of the operation"

File: docs/workflows/data/google-big-query.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    - [Update](https://cloud.google.com/bigquery/docs/reference/rest/v2/tables/update)
+    - [Patch](https://cloud.google.com/bigquery/docs/reference/rest/v2/tables/patch)
+    - [Delete](https://cloud.google.com/bigquery/docs/reference/rest/v2/tables/delete)
+- [Tabledata](https://cloud.google.com/bigquery/docs/reference/rest/v2/tabledata)
+    - [Insert All](https://cloud.google.com/bigquery/docs/reference/rest/v2/tabledata/insertAll)
+    - [List](https://cloud.google.com/bigquery/docs/reference/rest/v2/tabledata/list)
+
+### Request Config
+
+The available options here will change depending on the selected resource and action. For example, if the selected action creates or updates a resource then a request body is required that contains the new or updated resource. Some actions require a dataset ID, some require both a dataset ID and a table ID, and some require neither. There are also some options that are only relevant to a specific resource & action, such as the "include hidden datasets" option when "Datasets: List" is selected).
```
Comment (cephirothdy2j) [inline docs/workflows/data/google-big-query.md]: Change the ampersand (&) to "and"; it's considered bad form to use an ampersand anywhere that is not a proper noun or in a space-constricted headline.

Also. this paragraph ends with a closing parenthesis, but there is no corresponding opening character.

Finally, can you include one example resource/action here with a screenshot of the edit panel and a short explanation? e.g. Tabledata > List

File: docs/workflows/data/google-big-query.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+### Result
+
+Optionally, you may specify a payload path at which to place the results. If the request succeeds but no result is returned by BigQuery (for example, when deleting a resource)  the result will be `{ success: true }`.
```
Comment (cephirothdy2j) [inline docs/workflows/data/google-big-query.md]: The result is not always optional; it depends on the resource/action selected.

File: docs/workflows/data/google-ml.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# GCP ML
```
Comment (cephirothdy2j) [inline docs/workflows/data/google-ml.md]: Same as BigQuery title comment; add "Node"

File: docs/workflows/data/google-big-query.md (hunk 2 above)
Comment (cephirothdy2j) [reply docs/workflows/data/google-big-query.md]: Can you add a link to "Google BigQuery" going to one of their pages with a general overview of what the service is? We do that for other nodes that interact with third-party services.

File: docs/workflows/data/google-ml.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+The GCP (Google Cloud Platform) ML node allows a workflow to send data to Google ML and store the resulting predictions on the workflow payload.
```
Comment (cephirothdy2j) [inline docs/workflows/data/google-ml.md]: Like with BigQuery, add a "Google ML" link to one of their pages explaining what the service is.

File: docs/workflows/data/google-ml.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+### Service Account Auth Token
+
+A [service account auth token](https://cloud.google.com/docs/authentication/getting-started#creating_a_service_account) is required for the workflow to authenticate with Google ML. You may enter this token one of two ways:
+
+- **JSON Template** accepts a [JSON template](/workflows/accessing-payload-data/#json-templates) for the token.
+- **Payload Path** accepts a [payload path](/workflows/accessing-payload-data/#payload-paths) for the token.
+
+### Cloud ML Model Configuration
+
+Specify the name of the model for which you want to get predictions. You may optionally specify the model version as well; if no version is provided the model's default version will be used.
```
Comment (cephirothdy2j) [inline docs/workflows/data/google-ml.md]: Provide a little more info on what a "model" is; or link to a page where Google describes it.

File: docs/workflows/data/google-ml.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+### Input
+
+The instances path is a payload path that points to the instances data you want to get predictions for.
```
Comment (cephirothdy2j) [inline docs/workflows/data/google-ml.md]: Provide a little more info on what an "instance path" is; or link to a page where Google describes it.

File: docs/workflows/outputs/google-pub-sub.md [hunk 10]
```diff
@@ -1,8 +1,8 @@
-# Google Pub/Sub
+# GCP Pub/Sub
```
Comment (cephirothdy2j) [inline docs/workflows/outputs/google-pub-sub.md]: Make the same changes you made for this file in the "Google Pub/Sub" trigger as well. (docs/workflows/triggers/google-pub-sub.md and the docs navigation)

**Review by cephirothdy2j — CHANGES_REQUESTED** [review]

**Review by cephirothdy2j — APPROVED** [review]

**Review by spacetc62 — APPROVED** [review]

**Pull request merged**